client-linuxapp/src/serviceinfo.rs

```diff
@@ -159,11 +288,138 @@ async fn process_serviceinfo_in(si_in: &ServiceInfo) -> Result<()> {
                     .with_context(|| format!("Error parsing rhsm {} value", key))?;
                 rhsm_perform_insights = Some(value);
             }
+        } else if module == "binaryfile" {
+            if key == "name" {
+                if binary_file_in_progress.path.is_some() {
+                    bail!(
+                        "Got binaryfile path {:?} after path {:?}",
+                        value,
+                        binary_file_in_progress.path
+                    );
+                }
+                binary_file_in_progress.path = Some(
+                    value
+                        .as_str()
+                        .context("Error parsing binaryfile name")?
+                        .to_string(),
+                );
+            } else if key == "length" {
+                if binary_file_in_progress.length.is_some() {
+                    bail!(
+                        "Got binaryfile length {:?} after length {:?}",
+                        value,
+                        binary_file_in_progress.length
+                    );
+                }
+                binary_file_in_progress.length =
+                    Some(value.as_u64().context("Error parsing binaryfile length")?);
+                binary_file_in_progress.contents = Some(Vec::with_capacity(
+                    binary_file_in_progress.length.unwrap() as usize,
+                ));
+            } else if key.starts_with("data") {
+                if binary_file_in_progress.contents.is_none() {
+                    bail!("Got binaryfile data before length {:?}", value);
+                }
+                binary_file_in_progress
+                    .contents
+                    .as_mut()
+                    .unwrap()
+                    .extend_from_slice(value.as_bytes().context("Error parsing binaryfile data")?);
+            } else if key == "mode" {
+                if binary_file_in_progress.mode.is_some() {
+                    bail!(
+                        "Got binaryfile mode {:?} after mode {:?}",
+                        value,
+                        binary_file_in_progress.mode
+                    );
+                }
+                binary_file_in_progress.mode =
+                    Some(value.as_u32().context("Error parsing binaryfile mode")?);
+            } else if key.starts_with("sha-") {
+                let sha_type = key.split('-').nth(1).unwrap();
+                let sha_value = value
+                    .as_bytes()
+                    .with_context(|| format!("Error parsing binaryfile sha-{} value", sha_type))?;
```

Comment:
`s/binaryfile/binary file/g`?

Reply:
Sure. Will push the fix after your first review pass finishes.